lib/child_process.js

```diff
@@ -761,20 +771,18 @@ function spawnWithSignal(file, args, options) {
   const child = spawn(file, args, opts);
 
   if (options && options.signal) {
+    const killSignal = sanitizeKillSignal(options.killSignal);
     // Validate signal, if present
     validateAbortSignal(options.signal, 'options.signal');
     function kill() {
-      if (child._handle) {
-        child._handle.kill(options.killSignal || 'SIGTERM');
-        child.emit('error', new AbortError());
-      }
+      abortChildProcess(child, killSignal);
     }
     if (options.signal.aborted) {
       process.nextTick(kill);
     } else {
-      options.signal.addEventListener('abort', kill);
+      options.signal.addEventListener('abort', kill, { once: true });
```

Comment:
Not a fan of the `once: true`, if you want this to be safer you can do `[kWeakHandler]: child` instead and save the two lines below of having to remember to remove the listener.

Reply:
I just added it for consistency as one of the calls had it, and the one in `spawn` didn't. I can just remove it, if it's preferred.